shared-libs/cht-datasource/src/libs/contact.ts

```diff
@@ -0,0 +1,16 @@
+import { Doc } from './doc';
+import { DataObject } from './core';
+
+interface DehydratedParent extends DataObject {
```

Comment:
question: What is the significance of `Dehydrated*`?

Reply:
Yeah, not totally happy with this name, but the "Hydrate" terminology comes from the [existing codebase](https://github.com/medic/cht-core/blob/899e30df5fc6b7c52c99893609887ae714c87cd0/shared-libs/lineage/src/hydration.js#L5010.  Basically, the parent lineage of contacts is normalized onto the contact docs in the DB so that each contact records the hierarchy of its parents/grandparents/etc.  But, instead of having each contact doc in the DB store the _whole data_ about the parent contact, we just normalize the `_id` and the `parent` fields.  However, when you want to get data about a contact and their parents, often you want to have all the data about the parent (and not just the normalized data).  The process of getting the rest of that parent data has been called "hydrating". 

Now that I have typed this all out, I am wondering if maybe "NormalizedParent" or something like that might be a better name for this interface (and a bit less specific to the historical context of the CHT...)